Superlib.js

```diff
@@ -0,0 +1,633 @@
+			item.creators.push({
+				lastName: person.trim().replace(/著/g, ''),
+				creatorType: 'author',
+				fieldMode: 1
+			});
+		} else if (person.endsWith('译')) {
+			item.creators.push({
+				lastName: person.trim().replace(/译/g, ''),
+				creatorType: 'translator',
+				fieldMode: 1
+			});
+		} else {
+			item.creators.push({
+				lastName: person,
+				creatorType: 'author',
+				fieldMode: 1
+			});
+		}
+	}
+}
+
+function doInventor(item, data) {
+	if (!data || data.length <= 0) return;
+	const persons = data.split(/，/g);
+	for (var person of persons) {
+		item.creators.push({
+			lastName: person,
+			creatorType: 'inventor',
+			fieldMode: 1
+		});
+	}
+}
+
+function doTag(item, data) {
+	if (!data || data.length <= 0) return;
+	const tags = data.split('；');
+	for (var tag of tags) {
+		item.tags.push(tag.trim());
+	}
+}
+
+function sourceType(doc) {
+	var type = doc.querySelector('.on span').textContent;
+	return type;
+}
+
+function detectType(doc) {
+	return TYPE_MAP[sourceType(doc)];
+}
+
+function detectWeb(doc, url) {
+	if (url.includes('/search')) {
+		return getSearchResults(doc, true) ? 'multiple' : false;
+	} else {
+		var dType = detectType(doc, url);
+		if (dType) {
+			return dType;
+		}
+	}
+	
+	return false;
+}
+
+function getSearchResults(doc, checkOnly, itemInfo) {
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll('.book1');
+	for (let row of rows) {
+		let a = row.querySelector('td>table .px14');
+		if (!a) {
+			a = row.querySelector('a');
+		}
+		if (!a) {
+			continue;
+		}
+
+		if (checkOnly) return true;
+		
+		let url = a.href;
+
+		// Z.debug(url);
+		let title = ZU.trimInternal(a.textContent);
+		
+		found = true;
+		if (!found['']) {
+			items[''] = '【提醒：存在失败的可能，请隔几个小时后试试，或单个抓取。】';
+		}
+		
+		if (itemInfo) {
+			var download = row.querySelector('.get a');
+			if (download && download.textContent == 'PDF下载') {
+				itemInfo[url] = {
+					pdfurl: download.href
+				};
+			}
+		}
+		if (title.startsWith('《') && title.endsWith('》')) {
+			title = title.replace(/《|》/g, '');
+		}
+		
+		items[url] = title;
+	}
+	return found ? items : false;
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		var itemInfo = {};
+		Zotero.selectItems(getSearchResults(doc, false, itemInfo), function (items) {
+			if (items) {
+				if (items['']) {
+					delete items[''];
+				}
+				scrapeItem(Object.keys(items), itemInfo);
+			}
+		});
+	} else {
+		scrape(doc, url);
+	}
+}
+
+function scrapeItem(items, itemInfo) {
+	ZU.processDocuments(items, function (doc, url) {
+		scrape(doc, url, itemInfo[url] ? itemInfo[url].pdfurl : null);
+	});
+}
+
+function scrape(doc, url, pdfurl) {
+	if (!url || url.length <= 0) {
+		return;
+	}
+	
+	var itemType = detectType(doc);
+	var item = new Zotero.Item(itemType);
+
+	item.url = url;
+	switch (sourceType(doc)) {
+		case '图书':
+			item.title = text(doc, '.tutilte');
+			var infos = text(doc, '.tubox dl');
+			infos = trim(infos);
```

Comment:
If you remove the shims from the top of the script, replace `trim(s)` with `ZU.trim(s)`. (`s.trim()` will still work.)